controlplane/pkg/nsmd/endpoints.go

```diff
@@ -46,41 +48,27 @@ type nsmEndpointServer struct {
 	nsmPodIPAddress    string
 }
 
-func (e nsmEndpointServer) AdvertiseEndpoint(ctx context.Context,
-	ar *nseconnect.EndpointAdvertiseRequest) (*nseconnect.EndpointAdvertiseReply, error) {
-	logrus.Infof("Received Endpoint Advertise request: %s", ar.RequestId)
+func (es nsmEndpointServer) RegisterNSE(ctx context.Context,
+	request *registry.NetworkServiceEndpoint) (*registry.NetworkServiceEndpoint, error) {
+	logrus.Infof("Received RegisterNSE request: %v", request)
 
-	// Compose a new Network Service Endpoint object name from Request_id which is NSE's pod
-	// UUID and Network Service Name.
-	endpointName := ar.RequestId + "-" + ar.NetworkEndpoint.NetworkServiceName
 	// Check if there is already Network Service Endpoint object with the same name, if there is
 	// success will be returned to NSE, since it is a case of NSE pod coming back up.
-	ep := e.model.GetEndpoint(endpointName)
+	ep := es.model.GetEndpoint(request.EndpointName)
 	if ep != nil {
-		logrus.Warnf("Network Service Endpoint object %s already exists", endpointName)
-		return &nseconnect.EndpointAdvertiseReply{
-			RequestId: ar.RequestId,
-			Accepted:  true,
-		}, nil
+		return nil, fmt.Errorf("Network Service Endpoint object %s already exists", request.EndpointName)
```

Comment:
This doesn't seem to match your comments as it returns an error if we have already seen the NSE registration.